src/core/stream/orchestrator/stream_orchestrator.ts

```diff
@@ -172,11 +175,18 @@ export default function StreamOrchestrator(
       return EVENTS.activePeriodChanged(period);
     }));
 
+  const isManifestTerminated$ = fromEvent(manifest, "manifestUpdate").pipe(
+    map(() => manifest.isTerminated),
+    startWith(manifest.isTerminated),
+    distinctUntilChanged()
+  );
+
   // Emits an "end-of-stream" event once every PeriodStream are complete.
   // Emits a 'resume-stream" when it's not
-  const endOfStream$ = areStreamsComplete(...streamsArray)
-    .pipe(map((areComplete) =>
-      areComplete ? EVENTS.endOfStream() : EVENTS.resumeStream()
+  const endOfStream$ = combineLatest([areStreamsComplete(...streamsArray),
+                                      isManifestTerminated$])
+    .pipe(map(([areComplete, isManifestTerminated]) =>
+      (areComplete && isManifestTerminated) ? EVENTS.endOfStream() : EVENTS.resumeStream()
```

Comment:
Maybe we want to add some distinctUntilChanged somewhere to avoid needlessly sending multiple `resumeStream`/`endOfStream` events in a row as we depend on two values now